{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Production of the Artemisinin Precursor Dihydroartemisinic Acid in _Bacillus subtilis_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Introduction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1 Literature review of Artemisinin\n",
    "Artemisinin and its derivatives are isolated from the plant sweet wormwood (_Artemisia annua_), although some other plants also have production potential [4]. Artemisinin plays a core role in the fight against malaria, and is recommended for use in combination with a partner drug. The role of artemisinin (or its derivatives) in this combination therapy is to minimise the _Plasmodium_ (parasite conferring malaria) biomass in the blood of infected patients [14].\n",
    "\n",
    "Artemisinin (as seen in Figure 1) is a sesquiterpene lactone; it consists of three terpene units and further contains an ester that is part of a carbon ring structure. The exact mode-of-action of artemisinin is still debated, but it is thought to be connected to the endoperoxide bridge [5,18]. A two-step mode of action has been suggested. Firstly, activation of artemisinin (or derivatives) by iron, either from heme or in molecular form, causes formation of free radicals and alkylating intermediates. Secondly, reaction between these newly formed free radical species and membrane–bound proteins specifically associated with malaria, confers the anti-malarial function [5].\n",
    "\n",
    "<center><img src=\"figures/Artemisinin.png\"></center>\n",
    "​​<p style=\"text-align:center\"><b>Figure 1</b>: Structure of artemisinin. From Ikram et al. [9]\n",
    "</p>\n",
    "\n",
    "Production of artemisinin in the native host is derived from the general terpenoid biosynthesis. Farnesyl diphosphate (FPP) is converted to amorpha-4,11-diene, which is the substrate of a P450 (CYP71AV1) enzyme that oxidises the compound through several steps. The final step in the pathway, converting artemisinic acid to artemisinin, is non-enzymatic, and occurs through a spontaneous reaction catalysed by UV light and oxygen [9]. The biosynthetic pathway can be seen in Figure 2.\n",
    "\n",
    "<center><img src=\"figures/Pathway.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 2</b>: Production pathway for artemisinin. ADS originates from <i>S. cerevisiae</i>, whereas the rest of the enzymes originate from  <i>A. annua</i>. From Ikram et al. [9].</p>\n",
    "\n",
    "The global artemisinin market is experiencing significant growth, with a 19.1% compound annual growth rate (CAGR) projected until 2031. The market size was USD 64 million in 2021 and is expected to reach USD 367.3 million in 2031. This growth is driven by several factors, including increased access to artemisinin-based combination therapies (ACTs) in malaria-endemic regions and the development of novel antimalarial medications [1].\n",
    "\n",
    "Malaria remains a significant health concern in endemic regions, and climate change is expected to exacerbate the issue [1]. Despite progress in reducing malaria deaths, the number of cases has increased, further emphasising the importance of effective anti-malarial treatments [13]. The market is categorised by the type of artemisinin, either extracted from _A. annua_ or by semi-synthetic production. The market is still in its early stages due to a limited number of global manufacturers [1]. However, potential for growth and the global demand for effective anti-malarial treatments can attract new players to the market. \n",
    "\n",
    "Challenges affecting market growth include side effects of antimalarial drugs, the prevalence of counterfeit and substandard drugs, supply chain disruptions, programmatic uncertainties, and a demand-supply mismatch. Despite these challenges, the market's growth potential is encouraged by increasing demand from malaria-endemic nations, along with increased R&D activities, improved medical infrastructure, and government initiatives for innovative anti-malarial medications [1]. Overall, the global artemisinin market is poised for substantial growth, driven by the pressing need for effective malaria treatment, especially in regions where malaria remains a significant public health issue.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.2 Literature review of the cell factory\n",
    "_Bacillus subtilis_ is a Gram-positive bacterium widely used for industrial production of both chemicals and proteins [24].  With multiple _B. subtilis_ processes generally recognized as safe (GRAS), it is a useful production host for many nutritional supplements and food additives [12]. Furthermore, with its lack of production of toxins [23], it is a good candidate for production of pharmaceuticals. Additionally, the bacterium is a fast grower, can grow from cheap substrates, and is known for its good secretion capabilities [24]. All of these properties make _B. subtilis_ a suitable cell factory used for larger scale production.\n",
    "\n",
    "Recent advances in synthetic biology have led to the development of new tools for genetic engineering of _B. subtilis_ [12]. These new tools can facilitate an easier integration of genes from the biosynthetic pathway of artemisinin and the subsequent optimization of expression likely required to achieve a high production. These advances make _B. subtilis_ a better choice as a production host compared to, for example, the commonly used cell factory _Escherichia coli_. _E. coli_ is capable of producing endotoxins in certain environments, depriving a lot of _E. coli_ processes of GRAS-status [16].\n",
    "\n",
    "Semisynthetic artemisinin production has already been developed in the yeast _Saccharomyces cerevisiae_ [21], however, this host has a slower growth rate than _B. subtilis_. Furthermore, an earlier precursor for artemisinin, amorphadiene, has been successfully produced in _B. subtilis_ at much higher yields than obtained previously in _S. cerevisiae_ [19]. Thus _B. subtilis_ composes a more attractive production host than _S. cerevisiae_. \n",
    "\n",
    "However, there are drawbacks to using _B. subtilis_. The genes required for artemisinin production come from a plant, which may cause _B. subtilis_ to struggle with functional production of these heterologous enzymes. Protein folding issues could arise in this new host as folding is not compartmentalised in prokaryotic systems. Furthermore, genes might not be expressed in feasible amounts [19]. These challenges need to be taken into account when utilising this host for semisynthetic artemisinin production."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Problem definition\n",
    "The critical anti-malarial drug, artemisinin, is derived from the sweet wormwood plant, where the precursor dihydroartemisinic acid (DHAA) is converted to artemisinin by UV light and oxygen (see Figure 2). However, production from plants is too time-consuming to meet global demand. To overcome this problem, engineered cell factories offer a promising solution, enabling rapid, scalable, and more environmentally friendly production of plant-products like artemisinin. Recent studies have shown successful production of the early artemisinin precursor, amorphadiene, using CRISPR-Cas9 in _B. subtilis_, making this a promising cell factory [19,22].\n",
    "\n",
    "This project focuses on engineering _B. subtilis_ as a cell factory to produce the late artemisinin precursor, DHAA. This is achieved by introducing the relevant genes into an existing genome scale metabolic (GSM) model generating a heterologous pathway. This is done utilising the following methods: computational modelling and simulations to design and optimise the cell factory, considering theoretical maximum yields, up- and downregulation targets, phenotypic phase planes and dynamic flux balance analysis. Moreover, gene knockout strategies are computed to identify genes that can be knocked out to improve DHAA production. Co-factor swapping for increased DHAA production is also investigated. The combination of these approaches aims to create an efficient and sustainable platform for DHAA production, for later conversion into artemisinin, addressing a crucial need in global healthcare."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Selection and assessment of existing GSM model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Currently there are 10 GSM models for _B. subtilis_ (strain 168), all based on one of the two earliest published GSM models by Oh et al. 2007 [27] or Henry et al. 2009 [8]. An overview of the GSM models for _B. subtilis_ is gathered in the table below.\n",
    "\n",
    "| GSM models         | Genes | Reactions | Metabolites | Refs.                  | Description                                                                                                                                                                                                       |\n",
    "|--------------|-------|-----------|-------------|------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|\n",
    "| Center model | 534   | 563       | 456         | Goelzer et al. (2008)  | Manually curated metabolic, genetic, and regulatory networks of central metabolism using published data and expert knowledge.                                                                                     |\n",
    "| iYO844       | 844   | 1021      | 988         | Oh et al. (2007)       | An in silico (i) generated model based on genomic, biochemical, and physiological information and high-throughput phenotyping experiments.                                                                        |\n",
    "| iBsu1103     | 1103  | 1437      | 1139        | Henry et al. (2009)    | An in silico model based on SEED annotations  and fitting of estimated thermodynamic data with experimental data combined with a directionality prediction method.                                                |\n",
    "| iBsu1103v2   | 1103  | 1451      | 1156        | Tanaka et al. (2013)   | Systematic mapping of non-essential regions by deletion mutations fitted to iBsu1103 predict interval deletion outcomes led to improvement of the accuracy of iBsu1103.                                           |\n",
    "| iBsu1147     | 1147  | 1742      | 1456        | Hao et al. (2013)      | Model constructed from genomic and bibliomic data, the model iBsu1103 and subsequent modifications made in accordance with simulations related to biomass and ATP synthesis.                                      |\n",
    "| iBsu1144     | 1144  | 1955      | 1103        | Kocabaş et al. (2017)  | The model was constructed from thermodynamic analyses and elimination of unconnected-reactions in the renewed B. subtilis reaction network, BsRN-2016.                                                            |\n",
    "| eciYO844     | 844   | 1021      | 988         | Massaiu et al. (2019)  | An enzyme-constrained (ec) model made by integrating enzyme restrictions in iYO844 based on publicly available proteomics and enzyme kinetic parameters for central carbon metabolic reactions, the GECKO method. |\n",
    "| etiBsu1209   | 1209  | 1948      | 1595        | Bi et al. (2023)       | Updated version of iBsu1147 utilising machine learning tools to fill the gaps and additional integration of enzymatic constraints (e), thermodynamic constraints (t), and transcriptional regulatory networks.    |\n",
    "| ecBSU1       | 1155  | 3307      | 1459        | Wu et al. (2023)       | Updated version of iBsu1147 through gene-protein-reaction updates, biomass reaction standardization etc. and subsequent appliance of enzymatic constrains.                                                        |\n",
    "| iBB1018      | 1018  | 1577      | 1291        | Blázquez et al. (2023) | Constructed on the basis on iBsu1103v2 subjected to manual curation with updated biochemical and physiological knowledge and manual gap-filling analysis.                                                         | \n",
    "\n",
    "<p style=\"text-align:center\"><b>Table 1</b>: Overview of 10 <i>B. subtilis in silico</i> models. See reference list. </p>\n",
    "\n",
    "Despite the many GSM models presented, only models iYO844, iBsu1103, iBsu1147 and iBB1018 were publicly available and could be used for further Memote analysis to assess the quality of the models. The key results from Memote are summarised in Table 2 below.\n",
    "\n",
    "\n",
    "| Organism | Total score [%] | Stoichiometric Consistency [%] | Mass Balance [%] | Metabolites Connectivity [%] |\n",
    "|:--------:|:---------------:|:------------------------------:|:----------------:|:----------------------------:|\n",
    "|  iYO844  |        86       |               100              |       94.4       |              100             |\n",
    "| iBsu1103 |        34       |               100              |        0.0       |             99.6             |\n",
    "| iBsu1147 |        32       |               0.0              |       97.2       |             99.0             |\n",
    "|  iBB1018 |        71       |               0.0              |       92.4       |              100             |\n",
    "\n",
    "<p style=\"text-align:center\"><b>Table 2</b>: Overview of the results from the Memote analyses used for assessment of the models. </p>\n",
    "\n",
    "Of the four, only iYO844 is both mass balanced and is stoichiometric consistent in its reactions, which is necessary for further implementation and analysis. Therefore, iYO844 will be used as a GSM model to implement and optimise the DHAA production. The full results of the Memote analysis are saved as html files in  [0_memote_analysis.ipynb](0_memote_analysis.ipynb).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Computer-Aided Cell Factory Engineering\n",
    "\n",
    "#### Incorporation of Heterologous Genes in the iYO844 Model\n",
    "\n",
    "For the production of the artemisinin precursor DHAA in a heterologous host, it is necessary to find an metabolite which can be used as a starting point for the incorporation of a heterologous pathway.\n",
    "\n",
    "_B. subtilis_ is capable of producing the metabolite FPP exclusively as a part of the 2-C-Methyl-D-erythritol-4-phosphate (MEP) pathway. This was verified for the GSM model iYO844. To increase the supply of the starting point, FPP, of the heterologous pathway in _B. subtilis_, an extra reaction which converts geranyl diphosphate (GPP) to FPP is added [19]. The enzyme responsible for the reaction, farnesyl pyrophosphate synthase (FPPS), comes from _S. cerevisiae_. \n",
    "\n",
    "For production of DHAA in _B. subtilis_, the following enzymes should be incorporated into the GSM model: amorphadiene synthase (ADS), amorphadiene oxidase (CYP71AV1), alcohol dehydrogenase in combination with amorphadiene oxidase (ADH1_CYP71AV), artemisinic aldehyde double-bond reductase (DBR2), and aldehyde dehydrogenase 1 in combination with amorphadiene oxidase, which is used in two different reactions in the GSM model (ALDH1_CYP71AV1 and ALDH1_CYP71AV1_2) (see Figure 2). The ALDH1_CYP71AV1 enzyme responsible for converting dihydroartemisinic aldehyde into DHAA also converts artemisinic aldehyde into artemisinic acid within _A. annua_. To account for the loss of flux to DHAA, this reaction also needs to be incorporated into the GSM model with the annotation ALDH1_CYP71AV1_2. Enzymes responsible for the reactions from amorphadiene to DHAA come from _A. annua_, while the ADS enzyme responsible for converting FPP into amorphadiene comes from _S. cerevisiae_. \n",
    "\n",
    "After the implementation of the seven reactions in the GSM model, the production of DHAA was tested. The results showed a successful production of DHAA within the modified GSM model. Furthermore, the modified model’s quality was assessed through a Memote analysis, and showed no significant difference to the original model. The modified GSM model  is saved as [iYO844_modified.xml](iYO844_modified.xml) in the data folder, and the full analysis can be seen in ​[1_incorporation_of_heterologous_genes](/1_incorporation_of_heterologous_genes.ipynb).\n",
    "\n",
    "\n",
    "#### Maximum Theoretical Yield\n",
    "To optimise the model for higher production of DHAA, different suitable carbon sources were tested to investigate which would result in the highest maximum theoretical yield. The model is by default run with glucose as the carbon source, yielding a maximum theoretical yield of 0.214 mmol DHAA per mmol glucose (converted to 0.53 Cmol DHAA per Cmol glucose). \n",
    "\n",
    "Alternative carbon sources in the GSM model were investigated and their ability to produce DHAA was assessed. Only carbon sources resulting in a growth rate greater than zero was investigated further, and the production rate of DHAA from these was provided in mmol DHAA gDW $^{-1}$ h $^{-1}$. From this, the alternative carbon sources resulting in the top 20 highest production rates of DHAA were looked into (see Table 3). Especially maltotriose, maltose, and sucrose looked promising. These three alternative carbon sources were analysed due to the high growth rates and DHAA production rates they confer, and their ready availability to be used in fermentation over other carbon sources. The maximum theoretical yields of DHAA obtained from these carbon sources were compared to that of glucose. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Carbon</th>\n",
       "      <th>Growth</th>\n",
       "      <th>Production</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Maltotriose exchange</td>\n",
       "      <td>0.420673</td>\n",
       "      <td>1.296000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Raffinose exchange</td>\n",
       "      <td>0.419096</td>\n",
       "      <td>1.291143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Alpha L Arabinan exchange</td>\n",
       "      <td>0.340266</td>\n",
       "      <td>1.048286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cellobiose exchange</td>\n",
       "      <td>0.269319</td>\n",
       "      <td>0.829714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Maltose exchange</td>\n",
       "      <td>0.269319</td>\n",
       "      <td>0.829714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Trehalose exchange</td>\n",
       "      <td>0.269319</td>\n",
       "      <td>0.829714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Sucrose exchange</td>\n",
       "      <td>0.269319</td>\n",
       "      <td>0.829714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Dextrin exchange</td>\n",
       "      <td>0.267743</td>\n",
       "      <td>0.824857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Palatinose exchange</td>\n",
       "      <td>0.267743</td>\n",
       "      <td>0.824857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Starch exchange</td>\n",
       "      <td>0.267743</td>\n",
       "      <td>0.824857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Melibiose exchange</td>\n",
       "      <td>0.267743</td>\n",
       "      <td>0.824857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Lactose exchange</td>\n",
       "      <td>0.263013</td>\n",
       "      <td>0.810286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>N-Acetylneuraminate exchange</td>\n",
       "      <td>0.240404</td>\n",
       "      <td>0.752000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Oxidized glutathione exchange</td>\n",
       "      <td>0.182070</td>\n",
       "      <td>0.553623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>N-Acetyl-D-mannosamine exchange</td>\n",
       "      <td>0.173458</td>\n",
       "      <td>0.548000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>N-Acetyl-D-glucosamine exchange</td>\n",
       "      <td>0.160729</td>\n",
       "      <td>0.494571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AMP exchange</td>\n",
       "      <td>0.154005</td>\n",
       "      <td>0.470286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3'-AMP exchange</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>0.470286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Deoxyadenosine exchange</td>\n",
       "      <td>0.153388</td>\n",
       "      <td>0.425000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Adenosine exchange</td>\n",
       "      <td>0.152317</td>\n",
       "      <td>0.465429</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Carbon    Growth  Production\n",
       "0              Maltotriose exchange  0.420673    1.296000\n",
       "1                Raffinose exchange  0.419096    1.291143\n",
       "2         Alpha L Arabinan exchange  0.340266    1.048286\n",
       "3               Cellobiose exchange  0.269319    0.829714\n",
       "4                  Maltose exchange  0.269319    0.829714\n",
       "5                Trehalose exchange  0.269319    0.829714\n",
       "6                  Sucrose exchange  0.269319    0.829714\n",
       "7                  Dextrin exchange  0.267743    0.824857\n",
       "8               Palatinose exchange  0.267743    0.824857\n",
       "9                   Starch exchange  0.267743    0.824857\n",
       "10               Melibiose exchange  0.267743    0.824857\n",
       "11                 Lactose exchange  0.263013    0.810286\n",
       "12     N-Acetylneuraminate exchange  0.240404    0.752000\n",
       "13    Oxidized glutathione exchange  0.182070    0.553623\n",
       "14  N-Acetyl-D-mannosamine exchange  0.173458    0.548000\n",
       "15  N-Acetyl-D-glucosamine exchange  0.160729    0.494571\n",
       "16                     AMP exchange  0.154005    0.470286\n",
       "17                  3'-AMP exchange  0.153907    0.470286\n",
       "18          Deoxyadenosine exchange  0.153388    0.425000\n",
       "19               Adenosine exchange  0.152317    0.465429"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load the DataFrame from the CSV file\n",
    "pd.read_csv('data/top_20_carbon_sources.csv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p style=\"text-align:center\"><b>Table 3</b>: A list of the top 20 carbon sources identified based on optimising production of DHAA.</p>\n",
    "\n",
    "The yields in mmol/mmol obtained from the three alternative carbon sources were all greater than that obtained from glucose (see Figure 3). Maltotriose yielded 0.76 mmol DHAA per mmol maltotriose, and maltose and sucrose both yielded 0.48 mmol DHAA per mmol carbon source. However, when converting to Cmol/Cmol, it could be observed that the maximum theoretical yields obtained from maltotriose (0.63 Cmol DHAA per Cmol maltotriose), maltose (0.61 Cmol DHAA per Cmol maltose), and sucrose (0.61 Cmol DHAA per Cmol sucrose) were not significantly higher than that obtained from glucose (see Figure 4). Furthermore, increasing the boundary for the different carbon sources did not yield a significant change in the production rate of DHAA. Therefore, it was decided to proceed with glucose as a carbon source for DHAA production. The full analysis can be seen in ​​[2_maximum_theoretical_yield.ipynb](2_maximum_theoretical_yield.ipynb).\n",
    "\n",
    "<center><img src=\"figures/Maximum_Yield_from_Different_Carbon_Sources_mol.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 3</b>: Bar plot of the top four carbon source candidates in mmol-DHAA/mmol-C-source.</p>\n",
    "\n",
    "<center><img src=\"figures/Maximum_Yield_from_Different_Carbon_Sources_cmol.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 4</b>: Bar plot of the top four carbon source candidates in Cmol-DHAA/Cmol-C-source.</p>\n",
    "\n",
    "#### Phenotypic Phase Plane Analysis\n",
    "To get a comprehensive perspective on how variations in environmental factors impact optimal DHAA production rates and biomass growth rates on a global scale, phenotypic phase plane analysis was applied on the modified GSM model. First, the correlation between biomass growth and the production of DHAA was examined (see Figure 5).\n",
    "\n",
    "<center><img src=\"figures/ppp_DHAA_vs_biomass.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 5</b>: A phenotypic phase plane illustrating the correlation between biomass formation and DHAA production.</p>\n",
    "\n",
    "From Figure 5, a trade-off between growth of _B. subtilis_ and production of DHAA can be observed. This is expected as both biomass formation and the production of DHAA compete for the same metabolites, oxygen and glucose.\n",
    "\n",
    "<center><img src=\"figures/ppp_DHAA_vs_O2.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 6</b>: A phenotypic phase plane illustrating the correlation between oxygen consumption and DHAA production.</p>\n",
    "\n",
    "<center><img src=\"figures/ppp_biomass_vs_O2.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 7</b>: A phenotypic phase plane illustrating the correlation between oxygen consumption and biomass formation. </p>\n",
    "\n",
    "In terms of biomass growth rate and DHAA production vs oxygen uptake rate, respectively, it is clear that high amounts of oxygen inhibits the cell. This leads to a decrease in production of both DHAA and biomass. The optimal oxygen uptake rate for DHAA production is observed (see Figure 6) to be approx. -4 mmol O $_2$ gDW $^{-1}$ h $^{-1}$, whereas the optimal oxygen uptake rate for biomass formation is approx. -5 mmol O $_2$ gDW $^{-1}$ h $^{-1}$ (see Figure 7). Thus, in accordance with the negative correlation between DHAA production and biomass growth rate, the oxygen uptake needs to be approx. -4 mmol O $_2$ gDW $^{-1}$ h $^{-1}$ to achieve the highest possible DHAA production. The full analysis can be seen in [3_phenotypic_phase_planes.ipynb](3_phenotypic_phase_planes.ipynb).\n",
    "\n",
    "#### Flux Scanning based on Enforced Objective Flux Analysis\n",
    "Flux Scanning based on Enforced Objective Flux (FSEOF) analysis reveals reactions that can be altered in flux to enhance the production of a specific compound, here DHAA. This enhancement is achieved by progressively enforcing a flux increase through the objective function, the reaction ALDH1_CYP71AV1, which produces DHAA.\n",
    "\n",
    "<center><img src=\"figures/FSEOF Analysis.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 8</b>: Bar plot showing reactions that experience a change of flux from the FSEOF analysis.</p>\n",
    "\n",
    "Through the analysis, it was determined that 41 reactions experience a change of flux when increasing the flux through ALDH1_CYP71AV1 (see Figure 8). To identify target reactions for up- or downregulation, reactions with relative changes in flux below 80% were filtered away, and a list of 15 targets was generated (see Figure 9). \n",
    "\n",
    "<center><img src=\"figures/Rec_Change_Flux_Above_80.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 9</b>: Plot of reactions with relative change in flux exceeding 80%. FPPS, DMATT, CYP71AV1, ADS and ALDH1_CYP71AV1 almost exactly overlap, CYTK1 and NDPK3 almost exactly overlap, and MECDPDH2, MEPCT, MECDPS, DXPS, CDPMEK, DMPPS and DXPRIi exactly overlap.</p>\n",
    "\n",
    "In Figure 9, what looks like 4 different curves are observed. However, this is because the fluxes through some of the reactions overlap, indicating that these reactions most likely are coupled in the model.\n",
    "The 13 reactions with the highest relative changes in flux were chosen as targets for up- or downregulation. The full analysis can be seen in [4_FSEOF_analysis.ipynb](4_FSEOF_analysis.ipynb).\n",
    "\n",
    "<b>Top 13 targets for up- or downregulation:</b>\n",
    "- <b>ALDH1_CYP71AV1</b> aldehyde dehydrogenase 1, amorphadiene oxidase (heterologous genes)\n",
    "- <b>CYP71AV1</b> amorphadiene oxidase (heterologous gene)\n",
    "- <b>ADS</b> amorphadiene synthase (heterologous gene)\n",
    "- <b>FPPS</b> farnesyl pyrophosphate synthase (heterologous gene)\n",
    "- <b>DMATT</b> Dimethylallyltranstransferase/farnesyl pyrophosphate synthase (_ispA_)\n",
    "- <b>MECDPDH2</b> 2-C-methyl-D-erythritol 2,4 cyclodiphosphate dehydratase (_ispG_)\n",
    "- <b>MEPCT</b> 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase (_ispD_)\n",
    "- <b>MECDPS</b> 2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase (_ispF_)\n",
    "- <b>DXPS</b> 1-deoxy-D-xylulose 5-phosphate synthase (_dxs_)\n",
    "- <b>CDPMEK</b> 4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase (_ispE_)\n",
    "- <b>DMPPS</b> 1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate reductase (_ispH_)\n",
    "- <b>DXPRIi</b> 1-deoxy-D-xylulose reductoisomerase (_ispC_)\n",
    "- <b>IPDDI</b> Isopentenyl-diphosphate D-isomerase (_idi_)\n",
    "\n",
    "#### Simulation of Batch Cultivations Using Dynamic Flux Balance Analysis\n",
    "To gain insight into how the modified GSM model behaves in a more dynamic perspective over time in a batch cultivation, a dynamic flux balance analysis (DFBA) was conducted. From Figure 10, the concentrations of biomass, glucose in the medium, and DHAA as functions of time can be observed. As expected, the concentration of biomass increases in a sigmoidal shape over time as a response to glucose consumption. The concentration of DHAA also increases over time with the consumption of glucose, but to a much lesser degree by a factor of 10 $^{-16}$ (seen from the green y-axis in Figure 10). The full analysis can be seen in [5_DFBA.ipynb](5_DFBA.ipynb).\n",
    "\n",
    "<center><img src=\"figures/DFBA_simulation.png\"></center>\n",
    "<p style=\"text-align:center\"><b>Figure 10</b>: DFBA plot illustrating the production of biomass and DHAA in relation to the consumption of glucose, running from t=0 to t=6. Awareness required regarding the factor 1e-16 with which to multiply the DHAA concentration.</p>\n",
    "\n",
    "#### Optimisation Target Analysis\n",
    "Together, OptGene and OptKnock were employed to identify potential targets for optimising product formation through gene knockouts and overexpression. However, when OptGene was applied to the modified GSM model, it yielded no optimization targets. OptKnock was applied, however due to its time-consuming nature and the risk of crashes, only reactions not hindering cell growth or DHAA production were considered as possible targets. Despite attempts to run the code on the HPC server for DTU, the OptKnock analysis remained incomplete. Literature was searched for possible manual knockout targets, but did not give a feasible result [17,20]. The full analysis can be seen in [6_Optimisation_target_analysis.ipynb](6_Optimisation_target_analysis.ipynb).\n",
    "\n",
    "#### Design for Co-factor Swapping\n",
    "It was investigated if DHAA productivity could be improved by swapping co-factors in other reactions, thereby increasing the availability of the specific co-factor that is involved in the production of DHAA. This hypothesis is supported by a paper [10] which showed that co-factor swapping in the central carbon metabolism has the potential to improve (theoretical) yields of some amino acids in _S. cerevisiae_.  \n",
    "\n",
    "Of particular interest for the target compound DHAA is the cofactor NADPH, which is used as reductive power in the CYP71AV1-catalysed reactions (see Figure 2). It was attempted to swap the co-factor specificity for NADPH to NADH in all other reactions than the CYP71AV1-catalysed reactions. However, this co-factor swap did not reveal any reactions that would lead to an increased theoretical yield of DHAA.\n",
    "\n",
    "Alternatively, it was investigated if swapping the co-factor specificity of the inserted CYP71AV1 from NADP+ to NAD+ and NADPH to NADH would allow an increased theoretical yield of DHAA. The relevant inserted heterologous reactions were manually rebuilt to utilise NADH instead of NADPH, and the analysis was conducted again. Unfortunately, this did not result in any potential targets. The full analysis can be seen in [7_Co-factor_swap.ipynb](7_Co-factor_swap.ipynb).\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Discussion\n",
    "\n",
    "The carbon source analysis showed that utilising an alternative carbon source to glucose - maltotriose, maltose, or sucrose - did not result in a significantly higher maximum theoretical yield of Cmol DHAA. Given the cost-effectiveness and availability of glucose as a fermentation carbon source, glucose was deemed the most feasible carbon source.\n",
    "\n",
    "The FSEOF analysis gave a list of the 13 reactions with the highest relative changes in flux and these were chosen as targets for up- or downregulation. A minimal change in flux through these reactions will confer the biggest change in flux through ALDH1_CYP71AV1. The FSEOF analysis showed that the MEP pathway enzymes encoded by _dxs_, _ispC_, _ispD_, _ispE_, _ispF_, _ispG_, _ispH_ and _ispA_ need to be upregulated to increase the flux through ALDH1_CYP71AV1. Artemisinin is a terpenoid, and in prokaryotes, the backbone of terpenoids is constructed through the MEP pathway. It is therefore reasonable that an upregulation of these enzymes would lead to an increased flux. Although the relative changes in flux through the enzymes mentioned above are high, their absolute fluxes are quite low. This is in accordance with the results from the phenotypic phase plane analysis which shows a negative correlation between the production of DHAA and the biomass growth rate, most likely because the intermediates in the MEP pathway are directed towards production of biomass instead of DHAA. Moreover, the analysis shows that the MEP enzyme IPDDI (an isomerase) should be downregulated to increase the flux through ALDH1_CYP71AV1. IPDDI catalyses the conversion of isopentenyl pyrophosphate (IPP) to dimethylallyl pyrophosphate (DMAPP), both serving as precursors for the backbone of all terpenoids. Downregulating this conversion could mean energy savings for the cell, thereby increasing the flux through just one of these precursors instead. Lastly, an upregulation of the heterologous reactions ALDH1_CYP71AV1, CYP71AV1, ADS, and FPPS will also increase the flux through ALDH1_CYP71AV1, logically, as all of these reactions precede the ALDH1_CYP71AV1 reaction. This will in turn produce more DHAA and ultimately artemisinin. \n",
    "\n",
    "The DFBA shows production of DHAA (although extremely low) in response to glucose consumption. However, the production of DHAA seems to reach a plateau before all glucose has been consumed. This can be rationalised as the cell transitions into the stationary phase where it redirects all flux through the MEP pathway towards other reactions essential for survival instead of production of DHAA.\n",
    "\n",
    "Although the utilised model in this project, iYO844, is of high quality, it is built on experimental data. This is an important factor when taking into account the obtained results as not all theoretical metabolites and reactions are necessarily a part of the model. Another factor is that amorphadiene oxidase (CYP71AV1) is used in three different reactions of the heterologous pathway. This complicates the analysis since the model does not know that all three reactions are controlled by the same enzyme. Furthermore, another important consideration is the constraints imposed by transferring genes between different domains of life. For instance, CYP71AV1 is a type of plant cytochrome P450 enzyme, which is required for production of many plant secondary metabolites. Inefficient production of this type of enzyme can result in suboptimal and unreliable production of plant metabolites in heterologous systems [6]. \n",
    "To summarise, further investigation into the possibility of utilising iYO844 for heterologous production of DHAA should be conducted.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Conclusion\n",
    "This project aimed to produce the artemisinin precursor, dihydroartemisinic acid (DHAA), for future large-scale production of the anti-malarial drug, artemisinin, utilising _B. subtilis_ as a cell factory. The _B. subtilis_ GSM model iYO844 was assessed using Memote, and seven relevant heterologous genes were inserted into the pathway to enable _in silico_ production of DHAA from a MEP pathway intermediate.\n",
    "\n",
    "Glucose was evaluated to be the best choice of carbon source for the production of DHAA. The FSEOF analysis showed that eight MEP pathway enzymes, and some of the heterologous genes, need to be upregulated to increase the flux through the ALDH1_CYP71AV1 reaction. Moreover, the analysis showed that the isomerase IPDDI, also from the MEP pathway, needs to be downregulated.\n",
    "\n",
    "The DFBA showed production of DHAA in response to glucose consumption although the production plateaus before the depletion of glucose. Furthermore, the phenotypic phase plane analysis showed a negative correlation between the production of DHAA and the biomass growth rate. \n",
    "\n",
    "Overall, this project illustrates that the production of the artemisinin precursor DHAA is possible in _B. subtilis_ _in silico_. However, further investigation is needed to implement this strategy on an industrial scale. If successful production of DHAA is achieved, it could assist in the global fight against malaria. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## References\n",
    "\n",
    "[1] Artemisinin market size, share, trends & forecast by, 2031. (n.d.). Business Research Insights | Global Market Research Report & Consulting. https://www.businessresearchinsights.com/market-reports/artemisinin-market-100155\n",
    "\n",
    "[2] Bi, X., Cheng, Y., Xu, X., Lv, X., Liu, Y., Li, J., Du, G., Chen, J., Ledesma‐Amaro, R., & Liu, L. (2023). EtiBsu1209: A comprehensive multiscale metabolic model for <i>Bacillus subtilis</i>. Biotechnology and Bioengineering, 120(6), 1623-1639. https://doi.org/10.1002/bit.28355\n",
    "\n",
    "[3] Blázquez, B., San León, D., Rojas, A., Tortajada, M., & Nogales, J. (2023). New insights on metabolic features of bacillus subtilis based on Multistrain genome-scale metabolic modeling. International Journal of Molecular Sciences, 24(8), 7091. https://doi.org/10.3390/ijms24087091\n",
    "\n",
    "[4] Discovery of qinghaosu (Artemisinin)—History of research and development of artemisinin-based antimalarials. (2018). Artemisinin-Based and Other Antimalarials, 1-67. https://doi.org/10.1016/b978-0-12-813133-6.00001-9\n",
    "\n",
    "[5] Du, G. (2018). Natural small molecule drugs from plants. Springer.\n",
    "\n",
    "[6] Gasser, B., Saloheimo, M., Rinas, U., Dragosits, M., Rodríguez-Carmona, E., Baumann, K., Giuliani, M., Parrilli, E., Branduardi, P., Lang, C., Porro, D., Ferrer, P., Tutino, M. L., Mattanovich, D., & Villaverde, A. (2008). Protein folding and conformational stress in microbial cells producing recombinant proteins: A host comparative overview. Microbial Cell Factories, 7(1). https://doi.org/10.1186/1475-2859-7-11\n",
    "\n",
    "[7] Hao, T., Han, B., Ma, H., Fu, J., Wang, H., Wang, Z., Tang, B., Chen, T., & Zhao, X. (2013). In silico metabolic engineering of bacillus subtilis for improved production of riboflavin, egl-237, (r,r)-2,3-butanediol and isobutanol. Molecular BioSystems, 9(8), 2034. https://doi.org/10.1039/c3mb25568a\n",
    "\n",
    "[8] Henry, C. S., Zinner, J. F., Cohoon, M. P., & Stevens, R. L. (2009). IBsu1103: A new genome-scale metabolic model of bacillus subtilis based on SEED annotations. Genome Biology, 10(6), R69. https://doi.org/10.1186/gb-2009-10-6-r69\n",
    "\n",
    "[9] Ikram, N. K., & Simonsen, H. T. (2017). A review of biotechnological Artemisinin production in plants. Frontiers in Plant Science, 8. https://doi.org/10.3389/fpls.2017.01966\n",
    "\n",
    "[10] King, Z. A., & Feist, A. M. (2014). Optimal cofactor swapping can increase the theoretical yield for chemical production in escherichia coli and saccharomyces cerevisiae. Metabolic Engineering, 24, 117-128. https://doi.org/10.1016/j.ymben.2014.05.009\n",
    "\n",
    "[11] Kocabaş, P., Çalık, P., Çalık, G., & Özdamar, T. H. (2017). Analyses of extracellular protein production in bacillus subtilis – I: Genome-scale metabolic model reconstruction based on updated gene-enzyme-reaction data. Biochemical Engineering Journal, 127, 229-241. https://doi.org/10.1016/j.bej.2017.07.005\n",
    "\n",
    "[12] Liu, Y., Liu, L., Li, J., Du, G., & Chen, J. (2019). Synthetic biology toolbox and chassis development in bacillus subtilis. Trends in Biotechnology, 37(5), 548-562. https://doi.org/10.1016/j.tibtech.2018.10.005\n",
    "\n",
    "[13] Malaria - WHO. (n.d.). World Health Organization (WHO). https://www.who.int/data/gho/data/themes/malaria\n",
    "\n",
    "[14] Malaria: Artemisinin partial resistance. (n.d.). World Health Organization (WHO). https://www.who.int/news-room/questions-and-answers/item/artemisinin-resistance\n",
    "\n",
    "[15] Massaiu, I., Pasotti, L., Sonnenschein, N., Rama, E., Cavaletti, M., Magni, P., Calvio, C., & Herrgård, M. J. (2019). Integration of enzymatic data in bacillus subtilis genome-scale metabolic model improves phenotype predictions and enables in silico design of poly-γ-glutamic acid production strains. Microbial Cell Factories, 18(1). https://doi.org/10.1186/s12934-018-1052-2\n",
    "\n",
    "[16] Nguyen, H. D., & Phan, T. T. (2022). A protocol to enhance soluble protein expression in the cytoplasm of bacillus subtilis. Methods in Molecular Biology, 233-243. https://doi.org/10.1007/978-1-0716-1859-2_14\n",
    "\n",
    "[17] OptGene: a framework for in silico metabolic engineering. (n.d.). ResearchGate. https://www.researchgate.net/publication/277216985_OptGene_a_framework_for_in_silico_metabolic_engineering\n",
    "\n",
    "[18] O’Neill, P. M., Barton, V. E., & Ward, S. A. (2010). The molecular mechanism of action of Artemisinin—The debate continues. Molecules, 15(3), 1705-1721. https://doi.org/10.3390/molecules15031705\n",
    "\n",
    "[19] Pramastya, H., Xue, D., Abdallah, I. I., Setroikromo, R., & Quax, W. J. (2021). High level production of amorphadiene using bacillus subtilis as an optimized terpenoid cell factory. New Biotechnology, 60, 159-167. https://doi.org/10.1016/j.nbt.2020.10.007\n",
    "\n",
    "[20] Predict gene knockout strategies — cameo 0.13.6 documentation. (n.d.). Cameo 0.13.6 documentation. https://cameo.bio/05-predict-gene-knockout-strategies.html\n",
    "\n",
    "[21] Ro, D., Paradise, E. M., Ouellet, M., Fisher, K. J., Newman, K. L., Ndungu, J. M., Ho, K. A., Eachus, R. A., Ham, T. S., Kirby, J., Chang, M. C., Withers, S. T., Shiba, Y., Sarpong, R., & Keasling, J. D. (2006). Production of the antimalarial drug precursor artemisinic acid in engineered yeast. Nature, 440(7086), 940-943. https://doi.org/10.1038/nature04640\n",
    "\n",
    "[22] Song, Y., He, S., Abdallah, I. I., Jopkiewicz, A., Setroikromo, R., Van Merkerk, R., Tepper, P. G., & Quax, W. J. (2021). Engineering of multiple modules to improve Amorphadiene production in Bacillus subtilis using CRISPR-cas9. Journal of Agricultural and Food Chemistry, 69(16), 4785-4794. https://doi.org/10.1021/acs.jafc.1c00498\n",
    "\n",
    "[23] Souza, C. C., Guimarães, J. M., Pereira, S. D., & Mariúba, L. A. (2021). The multifunctionality of expression systems in Bacillus subtilis: Emerging devices for the production of recombinant proteins. Experimental Biology and Medicine, 246(23), 2443-2453. https://doi.org/10.1177/15353702211030189\n",
    "\n",
    "[24] Su, Y., Liu, C., Fang, H., & Zhang, D. (2020). Bacillus subtilis: A universal cell factory for industry, agriculture, biomaterials and medicine. Microbial Cell Factories, 19(1). https://doi.org/10.1186/s12934-020-01436-8\n",
    "\n",
    "[25] Tanaka, K., Henry, C. S., Zinner, J. F., Jolivet, E., Cohoon, M. P., Xia, F., Bidnenko, V., Ehrlich, S. D., Stevens, R. L., & Noirot, P. (2012). Building the repertoire of dispensable chromosome regions in bacillus subtilis entails major refinement of cognate large-scale metabolic model. Nucleic Acids Research, 41(1), 687-699. https://doi.org/10.1093/nar/gks963\n",
    "\n",
    "[26] Wu, K., Mao, Z., Mao, Y., Niu, J., Cai, J., Yuan, Q., Yun, L., Liao, X., Wang, Z., & Ma, H. (2022). EcBSU1: A genome-scale enzyme-constrained model of <em>Bacillus subtilis</em> based on the ECMpy workflow. https://doi.org/10.20944/preprints202211.0351.v1\n",
    "\n",
    "[27] Oh, Y., Palsson, B. O., Park, S. M., Schilling, C. H., & Mahadevan, R. (2007). Genome-scale reconstruction of metabolic network in bacillus subtilis based on high-throughput Phenotyping and gene essentiality data. Journal of Biological Chemistry, 282(39), 28791-28799. https://doi.org/10.1074/jbc.m703759200\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "project",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
